internal/test/environment/testenv.go

```diff
@@ -14,7 +15,63 @@ func Setup() error {
 	if dockerHost == "" {
 		return errors.New("$TEST_DOCKER_HOST must be set")
 	}
-	return os.Setenv("DOCKER_HOST", dockerHost)
+	if err := os.Setenv("DOCKER_HOST", dockerHost); err != nil {
+		return err
+	}
+
+	if dockerCertPath := os.Getenv("TEST_DOCKER_CERT_PATH"); dockerCertPath != "" {
+		if err := os.Setenv("DOCKER_CERT_PATH", dockerCertPath); err != nil {
+			return err
+		}
+		if err := os.Setenv("DOCKER_TLS_VERIFY", "1"); err != nil {
+			return err
+		}
+	}
+
+	if kubeConfig := os.Getenv("TEST_KUBECONFIG"); kubeConfig != "" {
+		if err := os.Setenv("KUBECONFIG", kubeConfig); err != nil {
+			return err
+		}
+	}
+
+	if val := boolFromString(os.Getenv("TEST_REMOTE_DAEMON")); val {
+		if err := os.Setenv("REMOTE_DAEMON", "1"); err != nil {
+			return err
+		}
+	}
+
+	if val := boolFromString(os.Getenv("TEST_SKIP_PLUGIN_TESTS")); val {
```

Comment:
There's like a pattern here 😇, I think it could be factorized, at least between `TEST_KUBECONFIG`, `TEST_REMOTE_DAEMON` and `TEST_SKIP_PLUGIN_TESTS`